airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -61,14 +67,151 @@ public static Source sshWrappedSource() {
+
+    LOGGER.info("Prepared SQL query for TableFullRefresh is: " + preparedSqlQuery);
+    return queryTable(database, preparedSqlQuery);
+  }
+
+  @Override
+  public AutoCloseableIterator<JsonNode> queryTableIncremental(JdbcDatabase database,
+                                                               List<String> columnNames,
+                                                               String schemaName,
+                                                               String tableName,
+                                                               String cursorField,
+                                                               JDBCType cursorFieldType,
+                                                               String cursor) {
+    LOGGER.info("Queueing query for table: {}", tableName);
+    return AutoCloseableIterators.lazyIterator(() -> {
+      try {
+        final Stream<JsonNode> stream = database.query(
+            connection -> {
+              LOGGER.info("Preparing query for table: {}", tableName);
+
+              final String identifierQuoteString = connection.getMetaData()
+                  .getIdentifierQuoteString();
+              List<String> newColumnNames = handleFunctionsIfAnyInIdentifierList(database,
+                  columnNames, schemaName, tableName, identifierQuoteString);
+
+              final String sql = String.format("SELECT %s FROM %s WHERE %s > ?",
+                  handleFunctionsIfAnyInIdentifierList(connection, newColumnNames),
+                  sourceOperations
+                      .getFullyQualifiedTableNameWithQuoting(connection, schemaName, tableName),
+                  sourceOperations.enquoteIdentifier(connection, cursorField));
+              LOGGER.info("Prepared SQL query for queryTableIncremental is: " + sql);
+
+              final PreparedStatement preparedStatement = connection.prepareStatement(sql);
+              sourceOperations.setStatementField(preparedStatement, 1, cursorFieldType, cursor);
+              LOGGER.info("Executing query for table: {}", tableName);
+              return preparedStatement;
+            },
+            sourceOperations::rowToJson);
+        return AutoCloseableIterators.fromStream(stream);
+      } catch (final SQLException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  /**
+   * There is no support hierarchyid even in the native SQL Server jdbc driver in DataTypes The only
+   * way to get it as a text is to query it as "test_column.ToString() as test_column". So we need to
+   * make a pre-request to get actual types and then wrap required fields with toString() function
+   *
+   * @param database
+   * @param columnNames
+   * @param schemaName
+   * @param tableName
+   * @param enquoteSymbol
+   * @return the list with Column names updated to handle functions (if nay) properly
+   */
+  private List<String> handleFunctionsIfAnyInIdentifierList(JdbcDatabase database,
```

Comment:
The name of this method is a bit confusing. What about something like `getWrappedColumn`?

Reply:
Updated, thanks